code/game/machinery/colony_floodlights.dm

```diff
@@ -184,64 +189,90 @@
 				return
-			var/obj/item/tool/weldingtool/WT = I
+			var/obj/item/tool/weldingtool/welder = I
 
 			if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_ENGI))
 				to_chat(user, SPAN_WARNING("You have no clue how to repair [src]."))
-				return 0
+				return FALSE
 
 			if(repair_state == FLOODLIGHT_REPAIR_WELD)
-				if(WT.remove_fuel(1, user))
+				if(welder.remove_fuel(1, user))
 					playsound(loc, 'sound/items/weldingtool_weld.ogg', 25)
 					user.visible_message(SPAN_NOTICE("[user] starts welding [src]'s damage."),
 					SPAN_NOTICE("You start welding [src]'s damage."))
-					if(do_after(user, 40, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
-						if(QDELETED(src) || !WT.isOn() || repair_state != FLOODLIGHT_REPAIR_WELD)
+					if(do_after(user, 4 SECONDS, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
+						if(QDELETED(src) || !welder.isOn() || repair_state != FLOODLIGHT_REPAIR_WELD)
 							return
 						playsound(loc, 'sound/items/Welder2.ogg', 25, 1)
 						repair_state = FLOODLIGHT_REPAIR_CABLE
 						user.visible_message(SPAN_NOTICE("[user] welds [src]'s damage."),
 						SPAN_NOTICE("You weld [src]'s damage."))
-						return 1
+						return TRUE
 				else
 					to_chat(user, SPAN_WARNING("You need more welding fuel to complete this task."))
 			return TRUE
 
 		else if(iscoil(I))
-			var/obj/item/stack/cable_coil/C = I
+			var/obj/item/stack/cable_coil/coil = I
 			if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_ENGI))
 				to_chat(user, SPAN_WARNING("You have no clue how to repair [src]."))
-				return 0
+				return FALSE
 
 			if(repair_state == FLOODLIGHT_REPAIR_CABLE)
-				if(C.get_amount() < 2)
+				if(coil.get_amount() < 2)
 					to_chat(user, SPAN_WARNING("You need two coils of wire to replace the damaged cables."))
 					return
 				playsound(loc, 'sound/items/Deconstruct.ogg', 25, 1)
 				user.visible_message(SPAN_NOTICE("[user] starts replacing [src]'s damaged cables."),\
 				SPAN_NOTICE("You start replacing [src]'s damaged cables."))
-				if(do_after(user, 20, INTERRUPT_ALL, BUSY_ICON_GENERIC))
+				if(do_after(user, 2 SECONDS, INTERRUPT_ALL, BUSY_ICON_GENERIC))
 					if(QDELETED(src) || repair_state != FLOODLIGHT_REPAIR_CABLE)
 						return
-					if(C.use(2))
+					if(coil.use(2))
 						playsound(loc, 'sound/items/Deconstruct.ogg', 25, 1)
 						repair_state = FLOODLIGHT_REPAIR_SCREW
-						user.visible_message(SPAN_NOTICE("[user] starts replaces [src]'s damaged cables."),\
+						user.visible_message(SPAN_NOTICE("[user] replaces [src]'s damaged cables."),\
 						SPAN_NOTICE("You replace [src]'s damaged cables."))
 			return TRUE
 
+		else if(istype(I, /obj/item/device/lightreplacer))
+			if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_ENGI))
+				to_chat(user, SPAN_WARNING("You have no clue how to repair [src]."))
+				return FALSE
+
+			if(repair_state == FLOODLIGHT_REPAIR_UNSCREW)
+				to_chat(user, SPAN_WARNING("You need to unscrew [src]'s maintenance hatch."))
+				return FALSE
```

Comment:
Personally I feel like it would make more sense for the light replacer to work after the crowbar step, otherwise you're only unscrewing and not actually opening the maintenance hatch.
(This would add another couple of seconds onto the repair time though, so maybe some of the flavour text around that step could be changed instead?)

Reply:
I'd sooner delete the crowbar step entirely. The repair length is too long.

Reply:
Hm, yeah... If the crowbar step was changed to `"[user] starts removing [src]'s damaged lighting assembly."` (or something along those lines, similar to the light replacer text) rather than anything to do with a maintenance hatch, then I think that would solve it.

Reply:
Messages are tweaked.